session/bootstrap.go

```diff
@@ -722,3 +725,52 @@ func oldPasswordUpgrade(pass string) (string, error) {
 	newpass := fmt.Sprintf("*%X", hash2)
 	return newpass, nil
 }
+
+func fixNoneStates(store kv.Storage) error {
+	err1 := kv.RunInNewTxn(store, true, func(txn kv.Transaction) error {
+		m := meta.NewMeta(txn)
+		dbs, err := m.ListDatabases()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		for _, db := range dbs {
+			if db.Name.L == "mysql" {
+				err = fixNoneStateUsers(m, db)
+			}
+		}
+		return errors.Trace(err)
+	})
+	return errors.Trace(err1)
+}
+
+func fixNoneStateUsers(m *meta.Meta, db *model.DBInfo) error {
+	tbls, err := m.ListTables(db.ID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, tbl := range tbls {
+		switch tbl.Name.L {
+		case "db", "tables_priv", "columns_priv":
+			for _, col := range tbl.Columns {
+				if col.Name.L == "user" && col.State == model.StateNone {
+					log.Warnf("fix none state column %s.User", tbl.Name.O)
+					col.State = model.StatePublic
+					col.Offset = 2
+				}
+			}
+			for _, idx := range tbl.Indices {
+				for _, idxCol := range idx.Columns {
+					if idxCol.Name.L == "user" && idxCol.Offset == 0 {
+						log.Warnf("fix index column offset for %s.User", tbl.Name.O)
+						idxCol.Offset = 2
+					}
+				}
+			}
```

Comment:
Cloud we print all column offsets in the wrong table after we fixed?
It can be used to confirm that the table schema is correct.